File: auth/src/main/java/com/sherpa/auth/configuration/SecurityConfig.java
package com.sherpa.auth.configuration;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
import org.springframework.security.web.DefaultRedirectStrategy;
import org.springframework.security.web.SecurityFilterChain;
import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
import org.springframework.security.web.authentication.SimpleUrlAuthenticationSuccessHandler;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Configuration
@EnableWebSecurity
@RequiredArgsConstructor
public class SecurityConfig {

	@Bean
	public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
		http
			.csrf(AbstractHttpConfigurer::disable)
			.formLogin(AbstractHttpConfigurer::disable)
			.httpBasic(AbstractHttpConfigurer::disable)
			.authorizeRequests(authorizeRequests -> authorizeRequests
				.requestMatchers("/login", "/error", "/webjars/**").permitAll()
				.anyRequest().authenticated())
			.oauth2Login(oauth -> oauth
				.defaultSuccessUrl("/signInSuccess", true)
				.failureUrl("/signInFailure"));
		return http.build();
	}

	@Bean
	public AuthenticationSuccessHandler successHandler() {
		SimpleUrlAuthenticationSuccessHandler handler = new SimpleUrlAuthenticationSuccessHandler();
		handler.setRedirectStrategy(new DefaultRedirectStrategy());
		handler.setDefaultTargetUrl("/dashboard");
		return handler;
	}
}

File: gateway/src/main/kotlin/com/sherpa/gateway/configuration/SecurityConfig.kt
package com.sherpa.gateway.configuration

import org.slf4j.LoggerFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.reactive.EnableWebFluxSecurity
import org.springframework.security.config.web.server.ServerHttpSecurity
import org.springframework.security.web.server.SecurityWebFilterChain
import org.springframework.security.web.server.authentication.RedirectServerAuthenticationFailureHandler
import org.springframework.security.web.server.authentication.RedirectServerAuthenticationSuccessHandler
import org.springframework.security.web.server.authentication.ServerAuthenticationFailureHandler


@EnableWebFluxSecurity
@Configuration
class SecurityConfig {
    private val logger = LoggerFactory.getLogger(javaClass)


    @Bean
    fun springSecurityFilterChain(http: ServerHttpSecurity): SecurityWebFilterChain {
        http
            .csrf { it.disable() }
            .formLogin { it.disable() }
            .httpBasic { it.disable() }
            .authorizeExchange {
                it.pathMatchers("/test", "/login", "/error", "/webjars/**").permitAll()
                it.anyExchange().authenticated()
            }
            .requestCache {
                it.disable()
            }
            .oauth2Login {
                it.authenticationSuccessHandler(successHandler())
                it.authenticationFailureHandler(failureHandler())
            }
        return http.build()
    }

    @Bean
    fun successHandler(): RedirectServerAuthenticationSuccessHandler {
        return RedirectServerAuthenticationSuccessHandler("/signin/success")
    }


    @Bean
    fun failureHandler(): ServerAuthenticationFailureHandler {
        return RedirectServerAuthenticationFailureHandler("/signin/failure")
    }

}